Hymns/Tags/TagStore.swift

```diff
@@ -60,6 +75,21 @@ class TagStoreRealmImpl: TagStore {
             action(favorite)
         }.toNotification()
     }
+
+    // MARK: Realm Query Functions
+
+    /** Can be used either with a value to specificially query for one tag or without the optional to query all tags besides favorites*/
+    func querySelectedTags(tagSelected: String?) -> Results<TagEntity> {
+        guard let specificTag = tagSelected else {
+            return realm.objects(TagEntity.self)
+        }
+        return realm.objects(TagEntity.self).filter(NSPredicate(format: "tags == %@", specificTag))
+    }
+
+    func queryTagsForHymn(hymnIdentifier: HymnIdentifier) -> Results<TagEntity> {
+        let convertedKey = TagEntity.createPrimaryKey(hymnIdentifier: hymnIdentifier, tag: "")
```

Comment:
Do you need this? I feel like you can just query all the tags and then filter by hymn type, hymn number, and query param

Reply:
I would still need a function like his to take the hymnIdentifier and convert into string values for hymnType, hymn number, and query param right? I could take convertedKey out and filer on hymnIdentifier.hymnName but hymnType is not a string.

Reply:
Nevermind, I figured it out.

Reply:
Okay, I guess you can try it. I have doubts that it'll work, because you're looking for TagEntities with an empty string as its tag.

As for hymn type not being a string, you can use hymnType.abbreviatedValue. But you have to see how hymnType is being stored to determine what you should call. It may work, or you may need to change the createPrimaryKey function to call hymnType.abbreviatedValue to make it work

Reply:
It works as seen in my recent test. the reason is because the NSpredicate is using a CONTAINS for the filter. So even though they all have tags on the end of the string they still contain as part of their string this information.